### PR TITLE
adding behave test for get_source

### PR DESCRIPTION
**Pull request by ahmadnazeri**
# Description

### Select type(s) of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

# Checklist:

- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have fixed any merge conflicts

### CODE REVIEW

File: tests/end_to_end/features/dataframe.feature [hunk 1]
```diff
@@ -1,14 +1,15 @@
-Feature: Serving Pandas Dataframe with client.dataframe method 
+Feature: Serving Pandas Dataframe with client.dataframe method
+  
   Scenario Outline: Getting the Dataframe of a Transformation
     Given Featureform is installed
-    When I turn on autovariants
+    When I turn on autovariants
```
Comment (epps) [inline tests/end_to_end/features/dataframe.feature:5]: It looks like there's a whitespace here. Is this going to cause any issues?

File: tests/end_to_end/features/features_with_timestamp.feature [hunk 2]
```diff
@@ -1,15 +1,15 @@
 Feature: Features with Timestamp
   Scenario Outline: Registering Features with Timestamp Column
     Given Featureform is installed
-    When I turn on autovariants
+    When I turn on autovariants
```
Comment (epps) [inline tests/end_to_end/features/features_with_timestamp.feature:4]: Same whitespace question here.

File: tests/end_to_end/features/model_registration.feature [hunk 3]
```diff
@@ -1,7 +1,7 @@
 Feature: Model Registration
   Scenario Outline: Registering a model with both non-existing and existing feature and training set
     Given Featureform is installed
-    When I turn on autovariants
+    When I turn on autovariants
```
Comment (epps) [inline tests/end_to_end/features/model_registration.feature:4]: Whitespace?

File: client/src/featureform/grpc_client.py [hunk 4]
```diff
@@ -134,5 +133,15 @@ def _limited_traceback(limit):
         sys.tracebacklimit = original_limit
 
 
+def _extract_error_details(e: grpc.RpcError) -> status_pb2.Status:
```
Comment (aolfat) [inline client/src/featureform/grpc_client.py:136]: this is from oss

**Review by aolfat — APPROVED** [review]

**Review by epps — APPROVED** [review]
🙌

**Pull request merged**